### ISSUE TITLE
Timeline stopping on O5S

### ISSUE THREAD

**Issue by sactage** [quisquous/cactbot#33]
~90% of the time, hen I get sucked into a train car in O5S using raidboss, the timeline stops for the fight and doesn't come back up. Doesn't matter if it's the ghosts after add phase or the normal diabolic whistle ones.

**Comment by quisquous** [comment]
Thanks for the report.  I'll take a look at this when I get back into o5s.

**Comment by quisquous** [comment]
@xephero

**Comment by ezsoftware** [comment]
Had the same issue last night as well, but only once.  Another issue I noticed was getting out of the passenger car too slowly could cause the 2nd half of the fight to be out of sync.  Our WAR 2-shotted his mob and was out so fast that the whole 2nd half of the fight was off by like 12 seconds for me.

**Comment by quisquous** [comment]
Ahahaha.  Yeah, the o5s timeline doesn't have any syncs but the beginning.  It clearly needs something for when it comes back up.

The o5s timeline also restarts if it ever sees "Removing combatant Phantom Train".  I'm wondering if by going into the car, it removes some fake phantom train combatant for you which is triggering the timeline reset.  Can either of you look at your o5s logs in ACT (or re-import the log file from yesterday so you can see the encounter again), and see if that line appears anywhere other than on a wipe? Alternatively, can you look at the logs and let me know the hit points of the "real" phantom train?

e.g. for Byakko, there's a bunch of fake Byakkos and then one real one:
[22:54:50.020] 04:Removing combatant Byakko.  Max HP: 57250.
[22:54:50.020] 04:Removing combatant Byakko.  Max HP: 57250.
[22:54:50.020] 04:Removing combatant Byakko.  Max HP: 10997725.

**Comment by Ikuni17** [comment]
I added this to line 42 of the O5S timeline, which helped sync the second phase most of the time:

330 "Diabolic Headlamp"	sync /uses Diabolic Headlamp/ window 30,20

For some reason there was one time it didn't sync correctly and I don't exactly remember why. I have also had the issue of the timeline stopping.

I have a log with a 1% enrage from last night, never cleared though, and "Removing combatant Phantom Train" only appeared at the wipe. There were 22 results, with one real:

[21:59:27.401] 04:Removing combatant Phantom Train.  Max HP: 12860067.

**Comment by quisquous** [comment]
Ok, my theory was confirmed tonight.  The game decides that you're not in combat during the ghost section and so the timeline stops.  Maybe need to special case that zone and not consider you out of combat until there's a "Removing combatant Phantom Train".

**Comment by xephero** [comment]
For what it's worth, from the start of that clip, most of the abilities are outside the default 2.5s window to sync. The --sync-- line at 491 should be moved up to 495, that's just one mistiming.

**Comment by Ikuni17** [comment]
Since the timeline only breaks on certain runs, my guess at what is happening is this:

In that clip above, you can see that the train returns with a red name plate still because people (the tanks) can get back out before it comes back. However, if the adds in the train cars get buffed from someone getting captured, it can take longer for anyone to clear their adds and the Phantom Train will return before anyone is back out and have a yellow/unclaimed name plate. You can also see this if everyone is dead in the Phantom Train "area" and someone else is in a train car fighting a ghost; the Phantom Train will become unclaimed.

**Comment by quisquous** [comment]
https://github.com/quisquous/cactbot/blob/master/ui/raidboss/timeline.js#L606 is the line that is stopping the timeline.  I think either I need to add support for trigger files to opt out of auto-stopping timelines (and providing regexes to stop themeselves), or alternatively do it at an even higher level (in fights.js maybe) so that oopsy could use the same information as raidboss.

I should be able to take a look at it this weekend.

**Comment by quisquous** [comment]
Tried reproing this on O5N, but it didn't appear to happen, at least on my run.  However, I think I have a fix for it that I can push shortly and should work.

**Comment by ezsoftware** [comment]
still having the sync issue coming out of the passenger car on O5S

In savage, I think the timings are different, the phantom train returns as soon as the first person comes out of the car, so the timings can be off.  I think it needs to re-sync to the first few actions after passenger car portion.  Those are less important to be off, but it gets very important to be sync'd properly before the pushy ghost if anyone tries to trust the timings for when to move for that.

**Comment by quisquous** [comment]
The fight didn't stop though, correct? It just didn't sync properly?

**Comment by ezsoftware** [comment]
correct, I haven't seen the fight stop issue since the first week.

**Comment by quisquous** [comment]
Let me know if you see it again after the most recent changes I uploaded.

**Comment by ezsoftware** [comment]
Will let you know next Tuesday.

**Comment by ezsoftware** [comment]
Your fixes to the O5S timeline worked perfectly for me this week.
